backend/frontend/coursecreation/webapi.js

```diff
@@ -0,0 +1,224 @@
+import $ from 'jquery';
+
+var URL = 'http://localhost:5000';
+
+//uses the backend API to create a node
+function createNode(courseId, contents, renderer){
+  let node_value = {
+    contents,
+    renderer
+  };
+
+  let endpoint = URL + `/${courseId}/node/add/`;
+  return Promise.resolve($.ajax(endpoint, {
+    method: "POST",
+    data: node_value,
+    dataType: "json"
+  }));
+}
+
+function getNode(courseId, nodeId){
+  let endpoint = URL + `/${courseId}/node/${nodeId}/`;
+  return Promise.resolve($.ajax(endpoint,{
+    method: "GET",
+    dataType: "json"
+  }));
+}
+
+function updateNode(courseId, node){
+  let endpoint = URL + `/${courseId}/node/update/${node.id}/`;
+  let data = {contents: node.contents, renderer: node.renderer, children: JSON.stringify(node.children)};
+  return Promise.resolve($.ajax(endpoint, {
+    method: "POST",
+    data: data,
+    dataType: "json"
+  }));
+}
+
+function deleteNode(courseId, nodeId: string){
+  let endpoint = URL + `/${courseId}/node/delete/${nodeId}/`;
+  return Promise.resolve($.ajax(endpoint, {
+    method: "POST",
+    dataType: "json"
+  }));
+}
+
+function addRoot(courseId, rootId){
+  rootId = rootId.toString();
+
+  let endpoint = URL + `/${courseId}/root/set/${rootId}/`;
+  return Promise.resolve($.ajax(endpoint, {
+    method: "POST"
+  }));
+}
+
+function initializeCourse(courseName){
+  if(courseName === ""){
+    courseName = "CS130";
+  }
+
+  let endpoint = URL + `/0/course/add/`;
+  return Promise.resolve($.ajax(endpoint, {
+    method: "POST",
+    data: {name: courseName},
+    dataType: "json"
+  }));
+}
+
+function setPiazza(courseId, piazzaId){
```

Comment:
Sometimes piazza ID can be left empty (if the user doesn't type it). As it currently stands, this will error out if piazza ID isn't supplied.

Suggested fix: allow empty piazza IDs. If a piazza ID is supplied, then we can do the piazza integration. If it isn't supplied, then:

 - don't attempt to set piazza ID
 - don't attempt to fetch piazza posts
 - basically, ignore everything "piazza"